### ISSUE TITLE
Illegal reflective access operation (java 10)

### ISSUE THREAD

**Issue by vittali** [OpenHFT/Java-Runtime-Compiler#33]
I get the following error:

```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by net.openhft.compiler.CompilerUtils (file:/home/p/.gradle/caches/modules-2/files-2.1/net.openhft/compiler/2.3.0/8
72e7f90c0223c27fa63c25f22b3a6610fa1ae4f/compiler-2.3.0.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int)
WARNING: Please consider reporting this to the maintainers of net.openhft.compiler.CompilerUtils
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```

Here is the stack:

```
java.lang.RuntimeException: java.lang.UnsupportedOperationException
	at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.handleExceptions(JavacTaskImpl.java:158)
	at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:96)
	at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:90)
	at net.openhft.compiler.CachedCompiler.compileFromJava(CachedCompiler.java:101)
	at net.openhft.compiler.CachedCompiler.loadFromJava(CachedCompiler.java:143)
	at net.openhft.compiler.CachedCompiler.loadFromJava(CachedCompiler.java:76)
	at robobrix.compile.lib.impl.F_.compile(F_.java:84)
	at robobrix.compile.lib.impl.F_.compile(F_.java:77)
	at robobrix.compile.lib.TestCompile.simple(TestCompile.java:23)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:538)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:760)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:460)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:206)
Caused by: java.lang.UnsupportedOperationException
	at java.compiler@10.0.1/javax.tools.JavaFileManager.listLocationsForModules(JavaFileManager.java:566)
	at jdk.compiler/com.sun.tools.javac.api.ClientCodeWrapper$WrappedJavaFileManager.listLocationsForModules(ClientCodeWrapper.java:397)
	at jdk.compiler/com.sun.tools.javac.code.ModuleFinder$ModuleLocationIterator.hasNext(ModuleFinder.java:135)
	at jdk.compiler/com.sun.tools.javac.code.ModuleFinder.scanModulePath(ModuleFinder.java:288)
	at jdk.compiler/com.sun.tools.javac.code.ModuleFinder.findAllModules(ModuleFinder.java:184)
	at jdk.compiler/com.sun.tools.javac.comp.Modules.getUnnamedModuleCompleter(Modules.java:1408)
	at jdk.compiler/com.sun.tools.javac.comp.Modules.setCompilationUnitModules(Modules.java:459)
	at jdk.compiler/com.sun.tools.javac.comp.Modules.enter(Modules.java:264)
	at jdk.compiler/com.sun.tools.javac.comp.Modules.initModules(Modules.java:230)
	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.initModules(JavaCompiler.java:1037)
	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:931)
	at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.lambda$doCall$0(JavacTaskImpl.java:100)
	at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.handleExceptions(JavacTaskImpl.java:142)
	... 31 more

```
❯ sdk current java

Using java version 10.0.1-oracle
```
```

**Comment by tomshercliff** [comment]
Hi @vittali - thanks for raising this.

This looks to be the same problem as #27 - which was already addressed for Java 9 but a new version was never released.

I have just released compiler-2.3.1 which should hit maven central very soon - please pick this up and retest.

Cheers,
Tom

**Comment by RobAustin** [comment]
@tomshercliff thanks Tom for doing this, @vittali I'm going to close this issue, please raise another issue if you have any problems with compiler-2.3.1
